### PR TITLE
Add optional "holder" semantics to global concurrency limits

### PR DESCRIPTION
**Pull request by abrookins**
Adds the ability to record an optional "holder" of a global concurrency lock when incrementing and decrementing.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/prefect/server/models/concurrency_limits_v2.py [hunk 1]
```diff
@@ -317,3 +341,79 @@
...
+                _holder_fifo[_id].remove(holder)
+        else:
+            # If we receive a decrement without a holder or for a holder we
+            # don't know about, drop the oldest holder from the FIFO queue. We
+            # know we decremented successfully, so the dict now has too many
+            # holders.
+            if active_slots < len(_limit_holders[_id]):
+                holder_to_drop = _holder_fifo[_id].popleft()
+                del _limit_holders[_id][holder_to_drop]
+            logger.warning("Decremented slots for unknown or null holder %s", holder)
```
[the remark this answers is not in the document]
Comment (abrookins) [reply src/prefect/server/models/concurrency_limits_v2.py:405]: Real talk

File: docs/3.0rc/api-ref/rest-api/server/schema.json [hunk 2]
```diff
@@ -16696,19 +16732,6 @@
                         "title": "Schedules",
                         "description": "A list of schedules for the deployment."
                     },
-                    "concurrency_limit": {
```
Comment (abrookins) [inline docs/3.0rc/api-ref/rest-api/server/schema.json:16699]: I don't know why this is happening. 🤔

**Review by chrisguidry — APPROVED** [review]
These tests are outstanding!  I read over the decrement logic like 5 times and it makes sense to me.  It should always exit and I believe the strategy makes sense.

**Review by cicdw — APPROVED** [review]
Am I correct that this PR doesn't begin populating the holder field yet? Otherwise LGTM

**Comment by abrookins** [comment]
@cicdw Yeah, exactly. Here, we're starting to support clients populating limit holders when incrementing and decrementing limits, but we don't use or populate them yet for anything.

**Pull request merged**